whitelabelco/whitelabel-components
# Add workflow for code quality

## Description
This PR:
- Adds a Github Action Workflow to do a clean installation of node dependencies, cache/restore them, lint, test and build the source code
- The workflow is run on any push to `main` or `release/*` branches, and for all PRs (for any branch)
- Some updates to the yarn.lock file have occurred, as when I was trying to run GH Actions the Action was failing owing to some missing dependencies of dependencies. I remove and re-installed the yarn.lock and node_modules, which seemed to fix the issues.

## To Test :mag: 
This is a bit awkward to test on the current branch, as the workflow is written to only run for pushes to main or release branches, or when opening a new PR from any branch; however, I'd suggest testing as follows:
* Go to `Actions` in the navigation bar on Github and you should see a workflow called `code-quality`. The most recent workflow run is passing, as shown below:
![Screenshot 2023-03-30 at 09 42 35](https://user-images.githubusercontent.com/65023744/228780445-716f9d98-b32e-432c-ab39-8395f3cadf2c.png)
* If you wanted, you could add in the following line (line 11) to the yaml file, and then make a push to the branch, which will then trigger the workflow to run on this branch (please remove the line afterwards and re-push your changes)
![Screenshot 2023-03-30 at 09 45 12](https://user-images.githubusercontent.com/65023744/228781144-e8f580c8-0141-4dd0-9fe6-d7cfcd372ddc.png)
* When this PR's code is eventually pushed to the release branch, check that the Github Actions run successfully again.


## Checklist :white_check_mark:
~~- [ ] I have re-assigned the Github ticket and moved the ticket on the project board.~~
~~- [ ] If I have added an ENV var or other secrets, I have updated Dashlane and the ReadMe (and any other applicable docs)~~

## Additional Details and Screenshots :art: \
* I've limited the Workflow to only run on Node `19.x`. I hope this seems reasonable but please let me know if you think it best to include any other Node versions.
